skyportal/models.py

```diff
@@ -134,10 +142,17 @@ class Source(Base):
     tns_name = sa.Column(sa.Unicode, nullable=True)
 
     groups = relationship('Group', secondary='group_sources')
+    candidate_groups = relationship("Group", secondary="group_candidates")
     comments = relationship('Comment', back_populates='source',
+                            primaryjoin="comments.c.source_id == sources.c.id",
                             cascade='save-update, merge, refresh-expire, expunge',
                             passive_deletes=True,
                             order_by="Comment.created_at")
+    candidate_comments = relationship("Comment", back_populates="candidate",
+                                      primaryjoin="comments.c.candidate_id == sources.c.id",
```

Comment:
Why should comments have both a source_id and a comment_id?

Reply:
source_id is a foreign key, and comment_id is the primary key for the comment...